geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/string/MSetDUnitTest.java

```diff
@@ -114,6 +119,33 @@ public void testMSet_concurrentInstancesHandleBucketMovement() {
                     .allSatisfy(value -> assertThat(value).startsWith("valueTwo"))));
   }
 
+  @Test
+  public void testMSet_isTransactional() {
```

Comment:
Because the reference for the RedisString is changing in the implementation. I was not able to get that exception being thrown that way after performing the MSET. That is why the check now makes sure it's not the same object reference after the mset happens. I am happy to know if there is any mechanism to get the exception thrown.

Reply:
The following test fails without your changes and passes with it, and demonstrates that the transaction is rolled back if an exception is encountered during the MSET operation:
```
  public static final String THROWING_CACHE_WRITER_EXCEPTION = "to be ignored";

  @Test
  public void mset_isTransactional() {
    IgnoredException.addIgnoredException(THROWING_CACHE_WRITER_EXCEPTION);
    String hashTag = "{" + clusterStartUp.getKeyOnServer("tag", 1) + "}";

    String key1 = hashTag + "key1";
    String value1 = "value1";
    jedis.set(key1, value1);

    String listKey = hashTag + "listKey";
    jedis.lpush(listKey, "1", "2", "3");

    String nonExistent = hashTag + "nonExistentKey";

    String throwingKey = hashTag + "ThrowingRedisString";
    String throwingKeyValue = "ThrowingRedisStringValue";

    jedis.set(throwingKey, throwingKeyValue);

    // Install a cache writer that will throw an exception if a key with a name equal to throwingKey is updated or created
    clusterStartUp.getMember(1).invoke(() -> {
      RedisClusterStartupRule.getCache()
          .<RedisKey, RedisData>getRegion(DEFAULT_REDIS_REGION_NAME)
          .getAttributesMutator()
          .setCacheWriter(new ThrowingCacheWriter(throwingKey));
    });

    String newValue = "should_not_be_set";

    assertThatThrownBy(
        () -> jedis.mset(key1, newValue, nonExistent, newValue, throwingKey, newValue))
            .hasMessage(SERVER_ERROR_MESSAGE);

    assertThat(jedis.get(key1)).isEqualTo(value1);
    assertThat(jedis.type(listKey)).isEqualTo("list");
    assertThat(jedis.exists(nonExistent)).isFalse();
    assertThat(jedis.get(throwingKey)).isEqualTo(throwingKeyValue);

    IgnoredException.removeAllExpectedExceptions();
  }

  private static class ThrowingCacheWriter implements CacheWriter<RedisKey, RedisData> {
    private final byte[] keyBytes;

    ThrowingCacheWriter(String key) {
      keyBytes = key.getBytes(StandardCharsets.UTF_8);
    }

    @Override
    public void beforeUpdate(EntryEvent<RedisKey, RedisData> event) throws CacheWriterException {
      if (Arrays.equals(event.getKey().toBytes(), keyBytes)) {
        throw new CacheWriterException(THROWING_CACHE_WRITER_EXCEPTION);
      }
    }

    @Override
    public void beforeCreate(EntryEvent<RedisKey, RedisData> event) throws CacheWriterException {
      if (Arrays.equals(event.getKey().toBytes(), keyBytes)) {
        throw new CacheWriterException(THROWING_CACHE_WRITER_EXCEPTION);
      }
    }

    @Override
    public void beforeDestroy(EntryEvent<RedisKey, RedisData> event) throws CacheWriterException {

    }

    @Override
    public void beforeRegionDestroy(RegionEvent<RedisKey, RedisData> event)
        throws CacheWriterException {

    }

    @Override
    public void beforeRegionClear(RegionEvent<RedisKey, RedisData> event)
        throws CacheWriterException {

    }
  }
```